pkg/helper/client.go

```diff
@@ -161,12 +163,20 @@ func UpdateCurrentState(client client.Client, nodeNetworkState *nmstatev1alpha1.
 	return nil
 }
 
-func ping(target string) (string, error) {
-	cmd := exec.Command("ping", "-c", "3", target)
-	var outputBuffer bytes.Buffer
-	cmd.Stdout = &outputBuffer
-	cmd.Stderr = &outputBuffer
-	return outputBuffer.String(), cmd.Run()
+func ping(target string, timeout int) (string, error) {
+	output := ""
+	return output, wait.PollImmediate(1*time.Second, 50*time.Second, func() (bool, error) {
+		cmd := exec.Command("ping", "-c", "1", target)
```

Comment:
You already implemented Eventually, maybe we can extend it with Consistently here inside?

Reply:
Let's check if we really needed, adding consistently is adding complexity that you have to debug in case of error, let me add a TODO